Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -4577,10 +4577,13 @@ void WebPageProxy::forceRepaint(CompletionHandler<void()>&& callback)
 
     m_drawingArea->waitForBackingStoreUpdateOnNextPaint();
 
-    sendWithAsyncReply(Messages::WebPage::ForceRepaint(), [this, protectedThis = Ref { *this }, callback = WTFMove(callback)] () mutable {
-        callAfterNextPresentationUpdate([callback = WTFMove(callback)] (auto) mutable {
+    sendWithAsyncReply(Messages::WebPage::ForceRepaint(), [weakThis = WeakPtr { *this }, callback = WTFMove(callback)] () mutable {
+        if (weakThis) {
```

Comment:
Nice, good job.

I'm scared now. I wonder how many other parts of WebPageProxy might keep it alive by mistake. :/ I guess we'll find out if it continues to leak.

This part of the change needs to be approved by an owner.

Reply:
My first thought was that we should change all the other strong references used for callbacks in this file to weak refs, to prevent this from happening again somewhere else.

Then I thought: perhaps we could instead try to ensure that the callbacks get deleted when the web process is destroyed? Nope, can't do that because the ref here is what keeps the web process alive.

We may need to squint at https://bugs.webkit.org/show_bug.cgi?id=241353#c4 more.

Reply:
> We may need to squint at https://bugs.webkit.org/show_bug.cgi?id=241353#c4 more.

Agreed. Perhaps some lint checks can help.

Reply:
> This part of the change needs to be approved by an owner.

Can you suggest right reviewer for this?

Reply:
Anyone listed in `Source/WebKit/Owners` can approve the change, but I've already tagged specific owners who are familiar with WebKit process model and might have suggestions.

Reply:
OK, you have owner approval from Alex.

That said, I'd say this merits either (a) similar WeakPtrs throughout the file, or else (b) a follow-up bug report as a reminder for others to investigate further. My worry is that other WebPageProxy messages elsewhere may trigger the same problem.

Reply:
Filed https://bugs.webkit.org/show_bug.cgi?id=241583